### PR TITLE
SDXL: EDM training

### PR DESCRIPTION
**Pull request by townwish4git**
# What does this PR do?

Adds # (feature)
> [Elucidating the Design Space of Diffusion-Based Generative Models](https://arxiv.org/pdf/2206.00364.pdf)
support EDM training

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [x] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [x] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

**Review by hadipash — APPROVED** [review]

**Comment by townwish4git** [comment]
modified VAE decode processing and relative configurations to support EDM-trained playground-v2.5

File: examples/stable_diffusion_xl/configs/inference/sd_xl_base_edm.yaml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        num_idx: 1000
+        quantize_c_noise: False
+        flip: False
+
+        weighting_config:
+          target: gm.modules.diffusionmodules.denoiser_weighting.EDMWeighting
+        scaling_config:
+          target: gm.modules.diffusionmodules.denoiser_scaling.EDMScaling
+        discretization_config:
+          target: gm.modules.diffusionmodules.discretizer.EDMDiscretization
```
Comment (Fzilan) [inline examples/stable_diffusion_xl/configs/inference/sd_xl_base_edm.yaml:30]: 虽然有默认设置，建议这里写上EDMDiscretization 3个关键参数的配置，方便用户找
params: 
sigma_min: 0.002
sigma_max: 80.0 
rho: 7.0

File: examples/stable_diffusion_xl/gm/models/diffusion.py [hunk 2]
```diff
@@ -41,6 +43,8 @@ def __init__(
         self.input_key = input_key
         self.no_cond_log = no_cond_log
         self.scale_factor = scale_factor
+        self.latents_mean = list(latents_mean)
+        self.latents_std = list(latents_std)
```
Comment (Fzilan) [inline examples/stable_diffusion_xl/gm/models/diffusion.py:47]: 这里不能直接设置 list(latents_mean)，如果yaml没有设置latents_mean or latents_std，即latents_mean is None, latents_std is None, list(None) 会报：*** TypeError: 'NoneType' object is not iterable

Comment (townwish4git) [reply examples/stable_diffusion_xl/gm/models/diffusion.py]: done

File: examples/stable_diffusion_xl/configs/inference/sd_xl_base_edm.yaml [hunk 3]
```diff
@@ -0,0 +1,121 @@
+version: SDXL-base-1.0
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/configs/inference/sd_xl_base_edm.yaml:1]: "sd_xl_base_edm.yaml" -> "sd_xl_base_edm_pg2.5.yaml"

File: examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_910b_diffusers_edm.yaml [hunk 4]
```diff
@@ -0,0 +1,151 @@
+version: SDXL-base-1.0
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_910b_diffusers_edm.yaml:1]: `sd_xl_base_finetune_910b_diffusers_edm.yaml` -> `sd_xl_base_finetune_910b_edm.yaml`
`sd_xl_base_finetune_910b_edm.yaml` -> `sd_xl_base_finetune_910b_edm_pg2.5.yaml`

**Review by zhanghuiyao — APPROVED** [review]

File: examples/stable_diffusion_xl/configs/inference/sd_xl_base_edm.yaml (hunk 3 above)
Comment (townwish4git) [reply examples/stable_diffusion_xl/configs/inference/sd_xl_base_edm.yaml]: done

**Pull request merged**